quarkus:dev not working with 2.8.2.Final

### Describe the bug

After upgrading quarkus to version 2.8.2.Final I get the following exception when starting mvn quarkus:dev

[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.8.2.Final:dev
Caused by: java.lang.NullPointerException
at io.quarkus.maven.DevMojo.addProject (DevMojo.java:741)
at io.quarkus.maven.DevMojo.newLauncher (DevMojo.java:1028)
at io.quarkus.maven.DevMojo$DevModeRunner. (DevMojo.java:858)

Looking at the code of DevMojo it is due to this variable being null and called upon on line 741:
final ArtifactSources sources = module.getSources();

Is this due to one of my maven modules not having sources or resources? Bit stuck here...

### Expected behavior

The development mode should start without any exceptions.

### Actual behavior

NullPointerException at io.quarkus.maven.DevMojo.addProject (DevMojo.java:741)

### How to Reproduce?

Reproducer:
[subpersub.zip](https://github.com/quarkusio/quarkus/files/8590776/subpersub.zip)

Steps to reproduce:
1. mvn clean install
2. cd subpersub-api
3. mvn quarkus:dev -e

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "11.0.14.1" 2022-02-08 LTS OpenJDK Runtime Environment Microsoft-31205 (build 11.0.14.1+1-LTS) OpenJDK 64-Bit Server VM Microsoft-31205 (build 11.0.14.1+1-LTS, mixed mode)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.8.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven home: C:\devenv\java\apache-maven-3.8.2 Java version: 11.0.14.1, vendor: Microsoft, runtime: C:\devenv\java\jdk-11.0.14.1+1 Default locale: en_US, platform encoding: Cp1252 OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

### Additional information

_No response_

--- a/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java
@@ -721,12 +721,16 @@ public class DevMojo extends AbstractMojo {
 
     private void addProject(MavenDevModeLauncher.Builder builder, ResolvedDependency module, boolean root) throws Exception {
 
+        if (!ArtifactCoords.TYPE_JAR.equals(module.getType())) {
+            return;
+        }
+
         String projectDirectory;
         Set<Path> sourcePaths;
         String classesPath = null;
         Set<Path> resourcePaths;
         Set<Path> testSourcePaths;
-        String testClassesPath;
+        String testClassesPath = null;
         Set<Path> testResourcePaths;
         List<Profile> activeProfiles = Collections.emptyList();
 
@@ -736,6 +740,10 @@ public class DevMojo extends AbstractMojo {
                 : null;
         final ArtifactSources sources = module.getSources();
         if (mavenProject == null) {
+            if (sources == null) {
+                getLog().debug("Local dependency " + module.toCompactCoords() + " does not appear to have any sources");
+                return;
+            }
             projectDirectory = module.getWorkspaceModule().getModuleDir().getAbsolutePath();
             sourcePaths = new LinkedHashSet<>();
             for (SourceDir src : sources.getSourceDirs()) {
@@ -766,28 +774,33 @@ public class DevMojo extends AbstractMojo {
         }
 
         final Path sourceParent;
-        if (sources.getSourceDirs() == null) {
-            if (sources.getResourceDirs() == null) {
-                throw new MojoExecutionException("The project does not appear to contain any sources or resources");
+        if (sourcePaths.isEmpty()) {
+            if (sources == null || sources.getResourceDirs() == null) {
+                throw new MojoExecutionException(
+                        "Local dependency " + module.toCompactCoords() + " does not appear to have any sources");
             }
             sourceParent = sources.getResourceDirs().iterator().next().getDir().toAbsolutePath().getParent();
         } else {
-            sourceParent = sources.getSourceDirs().iterator().next().getDir().toAbsolutePath().getParent();
+            sourceParent = sourcePaths.iterator().next().toAbsolutePath().getParent();
         }
 
-        Path classesDir = sources.getSourceDirs().iterator().next().getOutputDir().toAbsolutePath();
-        if (Files.isDirectory(classesDir)) {
-            classesPath = classesDir.toString();
-        }
-        Path testClassesDir = module.getWorkspaceModule().getTestSources().getSourceDirs().iterator().next().getOutputDir()
-                .toAbsolutePath();
-        testClassesPath = testClassesDir.toString();
-
+        Path classesDir = null;
         resourcePaths = new LinkedHashSet<>();
-        for (SourceDir src : sources.getResourceDirs()) {
-            for (Path p : src.getSourceTree().getRoots()) {
-                resourcePaths.add(p.toAbsolutePath());
+        if (sources != null) {
+            classesDir = sources.getSourceDirs().iterator().next().getOutputDir().toAbsolutePath();
+            if (Files.isDirectory(classesDir)) {
+                classesPath = classesDir.toString();
             }
+            for (SourceDir src : sources.getResourceDirs()) {
+                for (Path p : src.getSourceTree().getRoots()) {
+                    resourcePaths.add(p.toAbsolutePath());
+                }
+            }
+        }
+        if (module.getWorkspaceModule().hasTestSources()) {
+            Path testClassesDir = module.getWorkspaceModule().getTestSources().getSourceDirs().iterator().next().getOutputDir()
+                    .toAbsolutePath();
+            testClassesPath = testClassesDir.toString();
         }
 
         testResourcePaths = new LinkedHashSet<>();
@@ -807,13 +820,13 @@ public class DevMojo extends AbstractMojo {
                 resourcePaths.addAll(
                         build.getResources().stream()
                                 .map(Resource::getDirectory)
-                                .map(Paths::get)
+                                .map(Path::of)
                                 .map(Path::toAbsolutePath)
                                 .collect(Collectors.toList()));
                 testResourcePaths.addAll(
                         build.getTestResources().stream()
                                 .map(Resource::getDirectory)
-                                .map(Paths::get)
+                                .map(Path::of)
                                 .map(Path::toAbsolutePath)
                                 .collect(Collectors.toList()));
             }
@@ -821,10 +834,11 @@ public class DevMojo extends AbstractMojo {
 
         if (classesPath == null && (!sourcePaths.isEmpty() || !resourcePaths.isEmpty())) {
             throw new MojoExecutionException("Hot reloadable dependency " + module.getWorkspaceModule().getId()
-                    + " has not been compiled yet (the classes directory " + classesDir + " does not exist)");
+                    + " has not been compiled yet (the classes directory " + (classesDir == null ? "" : classesDir)
+                    + " does not exist)");
         }
 
-        Path targetDir = Paths.get(project.getBuild().getDirectory());
+        Path targetDir = Path.of(project.getBuild().getDirectory());
 
         DevModeContext.ModuleInfo moduleInfo = new DevModeContext.ModuleInfo.Builder()
                 .setArtifactKey(module.getKey())

--- a/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java
+++ b/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java
@@ -339,7 +339,7 @@ public class LocalProject {
                 .setBuildDir(getOutputDir());
 
         final Build build = (modelBuildingResult == null ? getRawModel() : modelBuildingResult.getEffectiveModel()).getBuild();
-        boolean addedNonTestSourceSets = false;
+        boolean addDefaultSourceSet = true;
         if (build != null && !build.getPlugins().isEmpty()) {
             for (Plugin plugin : build.getPlugins()) {
                 if (!plugin.getArtifactId().equals("maven-jar-plugin")) {
@@ -348,7 +348,7 @@ public class LocalProject {
                 if (plugin.getExecutions().isEmpty()) {
                     final DefaultArtifactSources src = processJarPluginExecutionConfig(plugin.getConfiguration(), false);
                     if (src != null) {
-                        addedNonTestSourceSets = true;
+                        addDefaultSourceSet = false;
                         moduleBuilder.addArtifactSources(src);
                     }
                 } else {
@@ -356,7 +356,7 @@ public class LocalProject {
                         DefaultArtifactSources src = null;
                         if (e.getGoals().contains(ArtifactCoords.TYPE_JAR)) {
                             src = processJarPluginExecutionConfig(e.getConfiguration(), false);
-                            addedNonTestSourceSets |= src != null;
+                            addDefaultSourceSet &= !e.getId().equals("default-jar");
                         } else if (e.getGoals().contains("test-jar")) {
                             src = processJarPluginExecutionConfig(e.getConfiguration(), true);
                         }
@@ -365,10 +365,11 @@ public class LocalProject {
                         }
                     }
                 }
+                break;
             }
         }
 
-        if (!addedNonTestSourceSets) {
+        if (addDefaultSourceSet) {
             moduleBuilder.addArtifactSources(new DefaultArtifactSources(ArtifactSources.MAIN,
                     Collections.singletonList(new DefaultSourceDir(getSourcesSourcesDir(), getClassesDir())),
                     collectMainResources(null)));

--- a/integration-tests/maven/src/test/resources-filtered/projects/multijar-module/hello/pom.xml
+++ b/integration-tests/maven/src/test/resources-filtered/projects/multijar-module/hello/pom.xml
@@ -10,17 +10,8 @@
 
     <dependencies>
         <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-resteasy</artifactId>
-        </dependency>
-        <dependency>
             <groupId>org.acme</groupId>
-            <artifactId>acme-beans</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>org.acme</groupId>
-            <artifactId>acme-beans</artifactId>
-            <classifier>other</classifier>
+            <artifactId>acme-intermediary</artifactId>
         </dependency>
     </dependencies>
 

--- a/integration-tests/maven/src/test/resources-filtered/projects/multijar-module/intermediary/pom.xml
+++ b/integration-tests/maven/src/test/resources-filtered/projects/multijar-module/intermediary/pom.xml
@@ -1,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project>
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.acme</groupId>
+        <artifactId>acme-parent</artifactId>
+        <version>${revision}.${changelist}${sha1}</version>
+    </parent>
+    <artifactId>acme-intermediary</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.acme</groupId>
+            <artifactId>acme-beans</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.acme</groupId>
+            <artifactId>acme-beans</artifactId>
+            <classifier>other</classifier>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>jar-jdk11</id>
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                        <configuration>
+                            <classesDirectory>\${project.build.outputDirectory}/jdk11</classesDirectory>
+                            <classifier>jdk11</classifier>
+                        </configuration>
+                    </execution>
+                </executions>
+                <configuration>
+                    <excludes>
+                        <exclude>jdk11/**</exclude>
+                    </excludes>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>

--- a/integration-tests/maven/src/test/resources-filtered/projects/multijar-module/intermediary/src/main/java/Dummy.java
+++ b/integration-tests/maven/src/test/resources-filtered/projects/multijar-module/intermediary/src/main/java/Dummy.java
@@ -1,0 +1,1 @@
+public class Dummy {}

--- a/integration-tests/maven/src/test/resources-filtered/projects/multijar-module/pom.xml
+++ b/integration-tests/maven/src/test/resources-filtered/projects/multijar-module/pom.xml
@@ -21,6 +21,7 @@
     </properties>
     <modules>
         <module>beans</module>
+        <module>intermediary</module>
         <module>hello</module>
         <module>runner</module>
     </modules>
@@ -86,6 +87,11 @@
             <dependency>
                 <groupId>org.acme</groupId>
                 <artifactId>acme-runner</artifactId>
+                <version>\${project.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.acme</groupId>
+                <artifactId>acme-intermediary</artifactId>
                 <version>\${project.version}</version>
             </dependency>
         </dependencies>

--- a/integration-tests/maven/src/test/resources-filtered/projects/multijar-module/runner/pom.xml
+++ b/integration-tests/maven/src/test/resources-filtered/projects/multijar-module/runner/pom.xml
@@ -47,14 +47,14 @@
                 <configuration>
                     <systemPropertyVariables>
                         <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                        <maven.home>${maven.home}</maven.home>
+                        <maven.home>\${maven.home}</maven.home>
                     </systemPropertyVariables>
                 </configuration>
             </plugin>
             <plugin>
-                <groupId>${quarkus.platform.group-id}</groupId>
+                <groupId>\${quarkus.platform.group-id}</groupId>
                 <artifactId>quarkus-maven-plugin</artifactId>
-                <version>${quarkus.platform.version}</version>
+                <version>\${quarkus.platform.version}</version>
                 <executions>
                     <execution>
                         <goals>
@@ -102,12 +102,12 @@
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-surefire-plugin</artifactId>
                         <configuration>
-                            <skipTests>${skipTests}</skipTests>
+                            <skipTests>\${skipTests}</skipTests>
                         </configuration>
                     </plugin>
                     <plugin>
                         <artifactId>maven-failsafe-plugin</artifactId>
-                        <version>${surefire-plugin.version}</version>
+                        <version>\${surefire-plugin.version}</version>
                         <executions>
                             <execution>
                                 <goals>
@@ -116,9 +116,9 @@
                                 </goals>
                                 <configuration>
                                     <systemPropertyVariables>
-                                        <native.image.path>${project.build.directory}/${project.build.finalName}-runner</native.image.path>
+                                        <native.image.path>\${project.build.directory}/\${project.build.finalName}-runner</native.image.path>
                                         <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
-                                        <maven.home>${maven.home}</maven.home>
+                                        <maven.home>\${maven.home}</maven.home>
                                     </systemPropertyVariables>
                                 </configuration>
                             </execution>

